### DISCUSSION TITLE
Issue with quotes in client and server components

### DISCUSSION THREAD

**Discussion by romanusname** [vercel/next.js#50204] [Help] [answered]
### Summary

I am using style element in next js.
I like <style> and not <style jsx>. It works well in both server and client components.

But I have a problem with quotes. On the server and on the client they are not the same.

My code:
```
export default function Home() {
  return (
      <>
        <main></main>
          <style>{`main{grid-template-areas:"a b c";}`}</style>
      </>
  )
}
```

Error in console:
```
Warning: Text content did not match.
Server: "main{grid-template-areas:&quot;a b c&quot;;}"
Client: "main{grid-template-areas:"a b c";}"
```

**How can I not get an error? What text to write in <style> so that there is no error?**

i tried like this:
```
<main></main>
<style>{`main{grid-template-areas:&quot;a b c&quot;;}`}</style>
```

New error:
```
`Warning: Text content did not match.
Server: "main{grid-template-areas:&amp;quot;a b c&amp;quot;;}"
Client: "main{grid-template-areas:&quot;a b c&quot;;}"`
```

I really need help. I have no idea how to fix this

**Comment by icyJoseph** [comment, marked as answer]
Try with: 

```jsx
 <style
        dangerouslySetInnerHTML={{
          __html: `main{grid-template-areas:"a b c";}`,
        }}
      />
```

I think what happens is that react-dom, stringifies the DOM by escaping some characters into HTML entities:

```jsx
import { renderToString } from "react-dom/server";

const result = renderToString(
  <>
    <main>hi</main>
    <style>{`main{grid-template-areas:"a b c";}`}</style>
  </>
);

console.log(result);
```

It logs:

```html
<main>hi</main><style>main{grid-template-areas:&quot;a b c&quot;;}</style> 
```

Make sure you understand the dangers of `dangerouslySetInnerHTML`, and think again if this is a good way forward. Perhaps just use styled-jsx, or CSS Modules.
